src/gui/systray.cpp

```diff
@@ -105,6 +105,11 @@ void Systray::create()
     }
     hideWindow();
     emit activated(QSystemTrayIcon::ActivationReason::Unknown);
+    for (const auto *folderMap : FolderMan::instance()->map()) {
```

Comment:
Sorry I guess I was being unclear, this is the right hand side of the `:` which should be wrapped into `qAsConst()` or stored in a `const auto` variable. Currently this might detach.

Reply:
Didn't think of detaching b/c we're receiving a QMap here but I get your point now. But: Why not making the map method return const then as `const OCC::Folder::Map map()` instead of "casting" it? As it's whole purpose is to essentially generate an iterator for an independent class datablock.

Reply:
Because then for any non const use you'd force a copy for nothing NRVO wouldn't kick in. Returning const data objects is the exception not the norm in most cases it's not what you want.

Rule of thumbs: non const -> const "good", const -> non const "bad". Reality is finer than that but it's a good heuristic. :-)

Reply:
And one more thing: yes, you'd hope you don't need the qAsConst of the intermediate const variable (at least personally I find that annoying nowadays). Unfortunately Qt containers are CoW, there was a good reason for that, it's less of a good reason now that move semantic appeared but they can't just be changed to be non-CoW now. So we end up dealing with that legacy...

Reply:
Yeah thats obvious, my point was that the map() really should be const imho. If you attempt to use map() to modify it, this is not how the current code was built to be used afaik.

But nvmd, for this particular snippet it doesn‘t make difference, rather style, I‘ll go add a qAsConst later..

Reply:
> Yeah thats obvious, my point was that the map() really should be const imho. If you attempt to use map() to modify it, this is not how the current code was built to be used afaik.

I think you're having a slight confusion here: it's perfectly fine to get the `map()` and modify it (and so shouldn't be prevented). It's a data type returned by value. I'd be worried if it was a pointer or a reference being returned. There's no risk of breaking the encapsulation here.

I'd even argue it's bad API design to try to make it harder to modify the object returned by `map()`. It'd be perfectly legitimate for instance to get the object returned by `map()` and then want to use `std::remove_if` on it (or whatever) to keep only the entries for which the key start with a given prefix (I'm making up the example but you get the idea).

Reply:
OK turns out qAsConst was implicitly deleted for these kinds of rvalues (in this case QMap). Pulled it out into an extra const auto now.

Reply:
Oh right, `qAsConst` shouldn't be used on rvalues indeed, my bad.